Lib/test/test_long.py

```diff
@@ -956,6 +956,15 @@ def test_huge_rshift_of_huge(self, size):
         self.assertEqual(huge >> (sys.maxsize + 1), (1 << 499) + 5)
         self.assertEqual(huge >> (sys.maxsize + 1000), 0)
 
+    @support.cpython_only
+    def test_small_ints_in_huge_calculation(self):
+        a = 2 ** SHIFT
+        b = 2 ** SHIFT - 1
+        c = 2 ** SHIFT + 1
+        self.assertIs(a - b, 1)
+        self.assertIs(c + (-a), 1)
```

Comment:
I would prefer to have "d = (-a)" and "self.assertIs(c + d, 1)" to make it even more explicit that you test a+b operator here.

nitpick: test a+b before a-b :-)